File: app/src/main/java/com/tft/purchase/ReminderReceiver.java
package com.tft.purchase;

import android.app.Notification;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.app.PendingIntent;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.Build;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.List;
import java.util.Locale;
import java.util.concurrent.TimeUnit;

public class ReminderReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        PurchaseDbHelper db = new PurchaseDbHelper(context);
        List<PurchaseDbHelper.DueItem> list = db.listDueItems();
        int overdue = 0, today = 0, soon = 0;
        int reminderDays = context.getSharedPreferences("tft_settings", Context.MODE_PRIVATE).getInt("reminder_days", 5);
        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd", Locale.TAIWAN);
        Date now = startOfDay(new Date());
        for (PurchaseDbHelper.DueItem d : list) {
            if (d.purchaseCompleted || d.itemCompleted || d.deliveryDate == null || d.deliveryDate.isEmpty()) continue;
            try {
                Date target = sdf.parse(d.deliveryDate);
                if (target == null) continue;
                long diff = TimeUnit.MILLISECONDS.toDays(target.getTime() - now.getTime());
                if (diff < 0) overdue++;
                else if (diff == 0) today++;
                else if (diff <= reminderDays) soon++;
            } catch (Exception ignored) {}
        }

        if (overdue + today + soon > 0) {
            NotificationManager nm = (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
            String channelId = "delivery_reminders";
            if (Build.VERSION.SDK_INT >= 26) {
                NotificationChannel channel = new NotificationChannel(channelId, "交貨提醒", NotificationManager.IMPORTANCE_HIGH);
                channel.setDescription("全益採購追蹤每日交貨提醒");
                nm.createNotificationChannel(channel);
            }
            Intent open = new Intent(context, MainActivity.class);
            open.putExtra("screen", "reminders");
            PendingIntent pi = PendingIntent.getActivity(context, 0, open, PendingIntent.FLAG_UPDATE_CURRENT | PendingIntent.FLAG_IMMUTABLE);
            String text = "已逾期 " + overdue + " 項｜今天 " + today + " 項｜即將交貨 " + soon + " 項";
            Notification.Builder b = Build.VERSION.SDK_INT >= 26 ? new Notification.Builder(context, channelId) : new Notification.Builder(context);
            b.setSmallIcon(R.drawable.ic_notification)
                    .setContentTitle("全益採購追蹤｜交貨雷達")
                    .setContentText(text)
                    .setStyle(new Notification.BigTextStyle().bigText(text + "\n點我查看廠商與品項明細"))
                    .setAutoCancel(true)
                    .setContentIntent(pi);
            nm.notify(8235678, b.build());
        }
        DriveBackupHelper.backupIfDueAsync(context);
    }

    private Date startOfDay(Date d) {
        try {
            SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd", Locale.TAIWAN);
            Date x = sdf.parse(sdf.format(d));
            return x == null ? d : x;
        } catch (Exception e) { return d; }
    }
}

File: quan-yi-ai-work-os/app/src/main/java/tw/com/tft/aiworkos/ReminderReceiver.kt
package tw.com.tft.aiworkos

import android.Manifest
import android.app.AlarmManager
import android.app.Notification
import android.app.NotificationManager
import android.app.PendingIntent
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.pm.PackageManager
import android.os.Build

object ReminderScheduler {
    fun schedule(context: Context, reminderId: Long, caseId: Long, atMillis: Long, message: String) {
        val intent = Intent(context, ReminderReceiver::class.java).apply {
            putExtra("reminder_id", reminderId)
            putExtra("case_id", caseId)
            putExtra("message", message)
        }
        val pi = PendingIntent.getBroadcast(
            context,
            reminderId.toInt(),
            intent,
            PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE
        )
        val am = context.getSystemService(Context.ALARM_SERVICE) as AlarmManager
        am.setAndAllowWhileIdle(AlarmManager.RTC_WAKEUP, atMillis, pi)
    }
}

class ReminderReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        if (Build.VERSION.SDK_INT >= 33 && context.checkSelfPermission(Manifest.permission.POST_NOTIFICATIONS) != PackageManager.PERMISSION_GRANTED) return
        val caseId = intent.getLongExtra("case_id", -1L)
        val message = intent.getStringExtra("message") ?: "有案件需要處理"
        val open = Intent(context, MainActivity::class.java).apply {
            flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TOP
            putExtra("open_case_id", caseId)
        }
        val openPi = PendingIntent.getActivity(
            context,
            (caseId and 0x7fffffff).toInt(),
            open,
            PendingIntent.FLAG_UPDATE_CURRENT or PendingIntent.FLAG_IMMUTABLE
        )
        val notification = Notification.Builder(context, "workos_reminders")
            .setSmallIcon(android.R.drawable.ic_dialog_info)
            .setContentTitle("全益 AI Work OS 提醒")
            .setContentText(message)
            .setStyle(Notification.BigTextStyle().bigText(message))
            .setContentIntent(openPi)
            .setAutoCancel(true)
            .build()
        val id = intent.getLongExtra("reminder_id", System.currentTimeMillis()).toInt()
        (context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager).notify(id, notification)
    }
}
